staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -151,34 +179,27 @@ type priorityLevelState struct {
 }
 
 // NewTestableController is extra flexible to facilitate testing
-func newTestableController(
-	informerFactory kubeinformers.SharedInformerFactory,
-	flowcontrolClient flowcontrolclient.FlowcontrolV1beta1Interface,
-	serverConcurrencyLimit int,
-	requestWaitLimit time.Duration,
-	obsPairGenerator metrics.TimedObserverPairGenerator,
-	queueSetFactory fq.QueueSetFactory,
-) *configController {
+func newTestableController(config TestableConfig) *configController {
 	cfgCtlr := &configController{
-		queueSetFactory:        queueSetFactory,
-		obsPairGenerator:       obsPairGenerator,
-		serverConcurrencyLimit: serverConcurrencyLimit,
-		requestWaitLimit:       requestWaitLimit,
-		flowcontrolClient:      flowcontrolClient,
-		priorityLevelStates:    make(map[string]*priorityLevelState),
-	}
-	klog.V(2).Infof("NewTestableController with serverConcurrencyLimit=%d, requestWaitLimit=%s", serverConcurrencyLimit, requestWaitLimit)
-	cfgCtlr.initializeConfigController(informerFactory)
+		name:                       config.Name,
+		clock:                      config.Clock,
+		queueSetFactory:            config.QueueSetFactory,
+		obsPairGenerator:           config.ObsPairGenerator,
+		finishHandlingNotification: config.FinishHandlingNotification,
+		asFieldManager:             config.AsFieldManager,
+		foundToDangling:            config.FoundToDangling,
+		serverConcurrencyLimit:     config.ServerConcurrencyLimit,
+		requestWaitLimit:           config.RequestWaitLimit,
+		flowcontrolClient:          config.FlowcontrolClient,
+		priorityLevelStates:        make(map[string]*priorityLevelState),
+	}
+	klog.V(2).Infof("NewTestableController %q with serverConcurrencyLimit=%d, requestWaitLimit=%s", cfgCtlr.name, cfgCtlr.serverConcurrencyLimit, cfgCtlr.requestWaitLimit)
+	cfgCtlr.configQueue = workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "priority_and_fairness_config_queue")
```

Comment:
if you're going non-standard, please comment with why the different times are more suitable.